### PR TITLE
set providerID

### LINKED ISSUE #1711: set providerID
**What would you like to be added**:
set kubelet's `--provider-id` (or the component config equivalent)

**Why is this needed**:
"Real" clusters typically set this, and tools may read it for identification.

NOTE: this may not be so trivial, given kubeadm not having per-node kubelet config ... xref: https://github.com/kubernetes-sigs/kind/issues/1424 / https://github.com/kubernetes-sigs/kind/issues/1270 => https://github.com/kubernetes/kubeadm/issues/1682

### PR DESCRIPTION
**Pull request by BenTheElder**
closes #1711 

```
$ kubectl describe no | grep -i provider
ProviderID:                  kind://docker/kind/kind-control-plane
```

### CODE REVIEW

File: pkg/cluster/internal/create/actions/config/config.go [hunk 1]
```diff
@@ -59,7 +60,16 @@ func (a *Action) Execute(ctx *actions.ActionContext) error {
 	// create kubeadm init config
 	fns := []func() error{}
 
+	// TODO(bentheelder): settle on a better abstraction for this
+	// While we don't want much code aware of the provider name,
+	// we do want to surface it in the node providerID field
+	providerName := "docker"
```
Comment (BenTheElder) [inline pkg/cluster/internal/create/actions/config/config.go:66]: really hesitant to encourage further reliance on this, but I think it makes sense to surface this data in the providerID field and I want to ship that.
We can replace this with something else later (maybe a public method that isn't officially part of the interface or something, need to think on that)

Comment (BenTheElder) [reply pkg/cluster/internal/create/actions/config/config.go:66]: Perhaps `String()`

Comment (amwat) [reply pkg/cluster/internal/create/actions/config/config.go:66]: Yeah, Stringer probably seems good enough.

Comment (BenTheElder) [reply pkg/cluster/internal/create/actions/config/config.go:66]: https://github.com/kubernetes-sigs/kind/pull/1809 done

**Pull request merged**